### PR TITLE
Alex3 to dev

### PR DESCRIPTION
**Pull request by yorkal**
This implements the space laser function to work in tandem with the new submarine unit.  Adds a new class and changes the board attack function .

### CODE REVIEW

File: src/main/java/cs361/battleships/models/Board.java [hunk 1]
```diff
@@ -10,6 +10,7 @@
 
 	@JsonProperty private List<Ship> ships;
 	@JsonProperty private List<Result> attacks;
+	@JsonProperty private Weapon spaceLaser = new Weapon();
```
Comment (jctrotter) [inline src/main/java/cs361/battleships/models/Board.java]: Nice, adding the weapon here allows for easier implementation of weapons in the future. Great job!

File: src/main/java/cs361/battleships/models/Weapon.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.Sets;
+import com.mchange.v1.util.CollectionUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+public class Weapon {
```
Comment (zhanzhid) [inline src/main/java/cs361/battleships/models/Weapon.java]: nice, add a weapon class increases the modularity of our code base for future projects!

**Comment by zhanzhid** [comment]
Nice job

**Comment by jctrotter** [comment]
Looks good to me, merging into dev.

**Pull request merged**